### DISCUSSION TITLE
does this approach to zustand have cons?

### DISCUSSION THREAD

**Discussion by arasrezaei** [pmndrs/zustand#1820] [Q&A] [answered]
hello there,
what are the cons of using zustand like this:

```
export const useSettings = create(
 (set) => ({
      setSettings: set,
      theme: THEME.DEFAULT,
      fontSize: FONT.DEFAULT
    });
```

and then in some component;

```
const {setSettings,theme} = useSettings();

const changeTheme = ()=> setSettings({theme: THEME.DARK});

return (
<div>
<div>{theme}</div>
<btn onClick={changeTheme} />
</div>
) 
```

**Comment by dbritto-dev** [comment, marked as answer]
@arasrezaei nothing but is way better to put that logic in your store instead. IMHO having `const { theme, setDarkTheme } = useSettings()` makes more sense. Remember that any business logic should be of the store itself, that's makes easier to test your business logic.

**Reply by arasrezaei** [reply]
> @arasrezaei nothing but is way better to put that logic in your store instead. IMHO having `const { theme, setDarkTheme } = useSettings()` makes more sense. Remember that any business logic should be of the store itself, that's makes easier to test your business logic.

Thanks for your answer.
I think your way of doing it is more appealing for lots of people, but I don't know why my mental model for it is to have a general setter, it's easier and simpler for me.

**Reply by dbritto-dev** [reply]
@arasrezaei yeah some people use to think on `useState` first but for this is way better to think on `useReducer`
